bin/download_minister.py

```diff
@@ -0,0 +1,165 @@
+        with open(OUTPUT_FILE) as json_file:
+            try:
+                existing_list = json.load(json_file)
+            except ValueError as e:
+                return False
+        json_file.close()
+
+    # generate a list of choices from external source
+    choices = download_xml_from_source()
+
+    # if existing list does not exist
+    if not existing_list and choices:
+        return set_status(choices)
+    # if crawled choices list does not exist
+    elif existing_list and not choices:
+        return set_status(existing_list)
+    # if both exist, compare both lists for changes
+    elif existing_list and choices:
+        for row in existing_list:
+            if row in choices:
+                # case: minister found in both lists
+                # update minister if existing list is different from choices list
+                if cmp(existing_list[row], choices[row]):
+                    existing_list[row]['precedence'] = choices[row]['precedence']
+                    for langCode in lang_codes:
+                        existing_list[row][langCode] = choices[row][langCode]
+                        existing_list[row]['honorific_title_' + langCode] = choices[row]['honorific_title_' + langCode]
+                        # case: same minister with new position in choices, then append new position to existing list
+                        for position in choices[row]['positions_' + langCode]:
+                            if not [d for d in existing_list[row]['positions_' + langCode]
+                                    if d['title_' + langCode] == position['title_' + langCode]]:
+                                existing_list[row]['positions_' + langCode].append(position)
+                            else:
+                                # case: same minister with same position title but other details changed, then update
+                                idx = next((index for (index, d) in enumerate(existing_list[row]['positions_' + langCode]) if d['title_' + langCode] == position['title_' + langCode]), None)
+                                existing_list[row]['positions_' + langCode][idx].update(position)
+                        # case: same minister with existing position not in choices then update end date as today
+                        for position in existing_list[row]['positions_' + langCode]:
+                            if not [d for d in choices[row]['positions_' + langCode]
+                                    if d['title_' + langCode] == position['title_' + langCode]]:
+                                position['end_date'] = date.today().strftime("%Y-%m-%dT%H:%M:%S")
+            else:
+                # case: minister found in existing list but now in new list, update end date as today
+                for langCode in lang_codes:
+                    for position in existing_list[row]['positions_' + langCode]:
+                        position['end_date'] = date.today().strftime("%Y-%m-%dT%H:%M:%S")
```

Comment:
we're running the script nightly or irregularly so storing the date down to seconds seems like too much precision

Reply:
the date from external source comes in this format so I kept it in the same format